### PR TITLE
Slim ci cd integration

### PR DESCRIPTION
**Pull request by TomiwaAribisala-git**
[Slim-CI-CD-Integration](https://github.com/slimtoolkit/slim/issues/371)
==================================================================

This PR is a solution to the issue above, which indicates how to integrate Slim in CI/CD environments such as Jenkins, Gitlab, GitHub Actions, CitcleCI etc.
===============

The ci-cd-integration directory shows a use case of integrating Slim across Jenkins and GitHub Actions, with workflow/pipelines examples for basic Nodejs app.
==============

@kcq kindly review, thanks.

### CODE REVIEW

**Comment by TomiwaAribisala-git** [comment]
Thanks for the review @kcq, I have appropriated the core repo REAMDE to reflect the CI/CD integrations section as documentation only, I have also removed the ci-cd-integration directory which isn't necessary for this PR, I await your feedback.

**Comment by TomiwaAribisala-git** [comment]
Thanks for the review @kcq, I have appropriated the core repo REAMDE to reflect the CI/CD integrations section as documentation only, I have also removed the ci-cd-integration directory which isn't necessary for this PR, I await your feedback.

Upon approval, I created an issue here--https://github.com/slimtoolkit/slim/issues/583, for creating a repo that has the ci/cd integration examples.

**Comment by TomiwaAribisala-git** [comment]
Hello @kcq, I am following up as regards this PR, i have signed off all commits and the DCO check is successful.

File: README.md [hunk 1]
```diff
@@ -634,8 +634,46 @@ docker run --volumes-from dcert -e DOCKER_HOST=$DOCKER_HOST -e DOCKER_TLS_VERIFY
...
+```
+
+#### Integrating Slim in Github Actions
+Github Actions has Runners running docker in the background, the following example steps highlight integrating slim in Github Actions using [Docker-Slim Github Action](https://github.com/marketplace/actions/docker-slim-github-action):
+- Building an Image and Executing Slim commands 
+```
+docker build -t node_alpine:${{github.run_number}} .
+```  
+```
+slim build --target node_alpine:${{github.run_number}} -tag slim-${{github.run_number}}
```
Comment (kcq) [inline README.md:675]: Not clear what both of these snippets represent... They look like regular CLI commands you'd run in your terminal. What do they have to do with Github Actions?

I expected to see a small paragraph or, at least, a sentence that describes how to use the Github Action you referenced. And I expected to see a Github Action code snippet like this (instructing users to insert something like this in their Github Action):

```
      # Slim it!
      - uses: kitabisa/docker-slim-action@v1
        env:
          DSLIM_HTTP_PROBE: false
        with:
          target: ${{ github.repository }}:latest
          tag: "slim"
```

File: README.md [hunk 2]
```diff
@@ -634,8 +634,46 @@ docker run --volumes-from dcert -e DOCKER_HOST=$DOCKER_HOST -e DOCKER_TLS_VERIFY
...
+- There are several flags that accept file paths (`--include-path-file`, `--compose-file`, `--http-probe-cmd-file`, etc). You need volume mount the location of the referenced paths or the file paths themselves when you use the containerized version of Slim because the Slim app container won't have accept to the referenced files otherwise.
 
-There are several flags that accept file paths (`--include-path-file`, `--compose-file`, `--http-probe-cmd-file`, etc). You need volume mount the location of the referenced paths or the file paths themselves when you use the containerized version of Slim because the Slim app container won't have accept to the referenced files otherwise.
+### CI/CD INTEGRATIONS
+
+#### Integrating Slim in Jenkins
+Given Jenkins is running as a containerized environment in a virtual machine(AWS EC2 Instance), the following example steps highlight integrating Slim in Jenkins: 
+- Make Docker commands available in Jenkins
+```
+docker run -p 8080:8080 -p 50000:50000 -d \
```
Comment (kcq) [inline README.md:645]: Where is this `docker run` command suppose to go?

File: README.md [hunk 3]
```diff
@@ -634,8 +634,46 @@ docker run --volumes-from dcert -e DOCKER_HOST=$DOCKER_HOST -e DOCKER_TLS_VERIFY
...
 
 ### Using `*-file` Flags
+- There are several flags that accept file paths (`--include-path-file`, `--compose-file`, `--http-probe-cmd-file`, etc). You need volume mount the location of the referenced paths or the file paths themselves when you use the containerized version of Slim because the Slim app container won't have accept to the referenced files otherwise.
 
-There are several flags that accept file paths (`--include-path-file`, `--compose-file`, `--http-probe-cmd-file`, etc). You need volume mount the location of the referenced paths or the file paths themselves when you use the containerized version of Slim because the Slim app container won't have accept to the referenced files otherwise.
+### CI/CD INTEGRATIONS
+
+#### Integrating Slim in Jenkins
```
Comment (kcq) [inline README.md:641]: Need more context in these instructions... Where's the information about Jenkinsfile and how it needs to be customized? It's not clear where the provided snippets are supposed to go.

I expected to see a Jenkinsfile snippet like this (with the info about where to put it and how to customize it):

```
        stage("Build and Slim Docker Image") {
            steps {
                script {
                    echo "building and slimming docker image..."
                    sh 'docker build -t node_alpine:latest'
                    sh 'slim build --target node_alpine:latest --tag node_alpine:slim-$BUILD_NUMBER'
                    }
                }
            }
        }
```

File: README.md (hunk 1 above)
Comment (TomiwaAribisala-git) [reply README.md]: Hello @kcq, thanks for the feedback, I have rewritten the entire CI/CD integrations section in the repo README, kindly review.

File: README.md (hunk 3 above)
Comment (TomiwaAribisala-git) [reply README.md]: Hello @kcq, thanks for the feedback, I have rewritten the entire CI/CD integrations section in the repo README, kindly review.

**Comment by TomiwaAribisala-git** [comment]
Hello @kcq, I intend to remind about reviewing this PR, thanks.

File: README.md (hunk 1 above)
Comment (kcq) [reply README.md]: Thanks a lot! Good to have the config file snippets for the CI/CD integrations.

**Pull request merged**